src/GovUk.Education.ExploreEducationStatistics.Data.Model.Tests/Repository/FootnoteRepositoryTests.cs

```diff
@@ -13,247 +15,105 @@ namespace GovUk.Education.ExploreEducationStatistics.Data.Model.Tests.Repository
+                    .GenerateList(1))
+                .GenerateList(3);
+            
+            var (subject1, subject2) = _subjectsGenerator
+                .ForIndex(0, s => s.SetFilters(subject1Filters))
+                .GenerateTuple2();
+
+            var subject1IndicatorGroup1 = _indicatorGroupGenerator
+                .WithIndicators(_indicatorGenerator.GenerateList(1))
+                .WithSubject(subject1)
+                .Generate();
+
+            var footnotes = _footnoteGenerator
+                .ForIndex(0, s => s
+                    .Set(f => f.Content, "Applies to subject 1")
+                    .SetSubjects(_ => ListOf(subject1)))
+                .ForIndex(1, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 1")
+                    .SetFilters(_ => ListOf(subject1Filters[0])))
+                .ForIndex(2, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 2 group 1")
+                    .SetFilterGroups(_ => ListOf(subject1Filters[1].FilterGroups[0])))
+                .ForIndex(3, s => s
+                    .Set(f => f.Content, "Applies to subject 1 filter 3 group 1 item 1")
+                    .SetFilterItems(_ => ListOf(subject1Filters[2].FilterGroups[0].FilterItems[0])))
+                .ForIndex(4, s => s
+                    .Set(f => f.Content, "Applies to subject 1 indicator 1")
+                    .SetIndicators(_ => ListOf(subject1IndicatorGroup1.Indicators[0])))
+                
+                // Footnote applies to subject 1 filter 1
+                // and subject 1 filter 2 group 1
+                // and subject 1 filter 3 group 1 item 1
+                // and subject 1 indicator 1
+                .ForIndex(5, s => s
+                    .Set(f => f.Content, "Applies to multiple attributes of subject 1")
+                    .SetFilters(_ => ListOf(subject1Filters[0]))
+                    .SetFilterGroups(_ => ListOf(subject1Filters[1].FilterGroups[0]))
+                    .SetFilterItems(_ => ListOf(subject1Filters[2].FilterGroups[0].FilterItems[0]))
+                    .SetIndicators(_ => ListOf(subject1IndicatorGroup1.Indicators[0])))
+                .ForIndex(6, s => s
+                    .Set(f => f.Content, "Applies to subject 2")
+                    .SetSubjects(_ => ListOf(subject2)))
+                .GenerateList();
+            
+            var releaseFootnotes = _releaseFootnoteGenerator
+                .ForInstance(s => s.Set(rf => rf.Release, release))
+                .ForInstance(s => s.Set(rf => rf.Footnote, (_, _, context) => footnotes[context.Index]))
+                .GenerateList(footnotes.Count);
+
+            var releaseSubjects = _releaseSubjectGenerator
+                .ForInstance(s => s.Set(rs => rs.Release, release))
+                .ForIndex(0, s => s.Set(rs => rs.Subject, subject1))
+                .ForIndex(1, s => s.Set(rs => rs.Subject, subject2))
+                .GenerateList();
```

Comment:
Same as other comments - could use extensions to roll this up:
```cs
var releaseSubjects = _releaseSubjectGenerator
    .WithRelease(release)
    .ForIndex(0, s => s.SetSubject(subject1))
    .ForIndex(1, s => s.SetSubject(subject2))
    .GenerateList(2);
```

Similar thing applies for other test cases.

Reply:
As per my other response, I figured you'd not want `With` and `Set` extension methods for the simpler properties as it risks the extension classes getting massive, but I'm personally happy to do it for readability.

One thing though - I find the need to have some methods prefixed with "With" and others with "Set" a little strange to read - any reason we can't just opt for one or the other?  I think that in general the "Withs" defer to the "Sets" behind the scenes anyway.

Reply:
Done ta